safe/impact_function_v4/impact_function.py

```diff
@@ -644,7 +652,10 @@ def prepare(self):
                 )
                 return PREPARE_FAILED_BAD_INPUT, message
 
+            self._provenance['exposure_layer'] = self.exposure.source()
```

Comment:
Ah I understand. Because the layers are re-assigned, then it won't be able to run the IF again. So, it's not ready and need to call prepare method.
In that case, I will just add another status, which indicate that the IF has a proper output from the analysis. It will be set to True only after call run method finished.